picker_wrapper_test.go

```diff
@@ -80,7 +80,10 @@ func (s) TestBlockingPick(t *testing.T) {
 	var finishedCount uint64
 	for i := goroutineCount; i > 0; i-- {
 		go func() {
-			if tr, _, err := bp.pick(context.Background(), true, balancer.PickInfo{}); err != nil || tr != testT {
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
```

Comment:
Etc throughout where we make contexts in a loop.  I think each test case should generally create a context very early in the test function and use it throughout.

Reply:
Extracted context creation from all loops.